Lesson7 1

JacinthaDev opened the pull request.

JacinthaDev committed:
lesson 6

JacinthaDev committed:
lesson 7_1 step 1

JacinthaDev committed:
lesson 7_1 step 2

JacinthaDev committed:
lesson 7_1 step 3

JacinthaDev committed:
lesson 7_1 questions

gacurl commented on lesson6-questions.txt:
1 - 4: excellent
5: Incorrect command provided. Ensure to use `bin/rails routes` for listing routes.
6 - 10: excellent.
Your answers are well-informed and demonstrate a strong understanding of Rails concepts

gacurl commented on lesson7-questions.txt:
your answers are well-detailed and accurate. Minor improvements could be made in specifying methods and clarifying some terms

gacurl commented on app/controllers/application_controller.rb:
Using find will raise an exception if the user is not found, which could be problematic if user data is corrupted or manipulated. Use find_by to handle these cases more gracefully

JacinthaDev replied:
I get the following error "Unsupported argument type: 1 (Integer)" when I use "find_by" instead of "find"

gacurl commented on app/controllers/users_controller.rb:
suggestion: Add error handling for logon and set_user. Ensure logon action is secure and only accessible to unauthenticated users

gacurl commented on app/views/forums/_form.html.erb:
suggest using `text_area` for description.

gacurl commented on config/routes.rb:
Routing setup is mostly correct but needs refinement to adhere to RESTful principles and Rails conventions.
Avoid using get routes for CRUD actions (create, update, destroy), which should be handled by resources.
Use RESTful conventions: update delete '/users/:id' to delete '/users/:id'.
Consider adding root paths for the posts and subscriptions controllers or use resources for full CRUD actions

JacinthaDev replied:
I tried doing so but changing the routes has caused errors in my application because my views are reliant on route helper names. Changing them to "resources" gets rid of many of the route helpers and causes my program to crash. This was the code given to us for the exercise

JacinthaDev committed:
Update _form.html.erb form.text_field to text_area

JacinthaDev committed:
Update users_controller.rb index

JacinthaDev committed:
Update posts_controller.rb check_access and post_params methods

JacinthaDev committed:
Update all answers in lesson7-questions.txt

JacinthaDev committed:
Update before_action :check_access from delete to destroy in posts_controller.rb

JacinthaDev committed:
Add security for edit, destroy, and logon to prevent errors

Added 2 before actions, one so that a user cannot edit another user, and another to make sure a user does not delete their account while still logged in. Also made it so another user cannot log on as another user before logging off.

JacinthaDev committed:
Change method for delete user path from users#delete to users#destroy

gacurl commented on app/controllers/users_controller.rb:
likely because find_by expects a hash with key-value pairs, not just an integer
`find_by` needs to know which column to search, so you must specify the column name (like `id`) along with the value (`params[:id]`).

please try `@user = User.find_by(id: params[:id])`

gacurl reviewed (approved):
keep growing - you're doing great!

JacinthaDev committed:
Update set_user method in users_controller.rb
